src/main/java/com/dsec/backend/service/IUserService.java

```diff
@@ -13,5 +16,11 @@ public interface IUserService {
 
 	UserInfoDTO login(LoginInfoDTO loginInfoDTO, HttpServletResponse response);
 
-	UserInfoDTO getUser(Jwt user);
+	ResponseEntity<?> getUser(int id);
+
+	ResponseEntity<?> getAllUsers() throws JsonProcessingException;
+
+	ResponseEntity<?> editUser(int id, UserDTO userDTO);
+
+	ResponseEntity<?> DeleteUserByID(Integer id);
```

Comment:
Creation of a response entity should reside in Controllers, not Services. Please keep the business/processing logic in the Service and API/Communication logic in Controllers. These methods should either return DTOs or Models or something else that is not directly related to response communication logic. Please look at this [tutorial](https://www.baeldung.com/spring-hateoas-tutorial).